### PR TITLE
Fix `Variable $x might not be defined.` issues detected by PHPStan

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | maybe
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Fixes 168 issues detected by PHPStan level 1. After this, it will remain only 64 issues. Once fixed, we would be able to activate PHPStan level 1 on CI.

### CODE REVIEW

File: src/AuthLDAP.php [hunk 1]
```diff
@@ -1458,23 +1458,16 @@ public static function displayLdapFilter($target, $users = true)
         }
         $config_ldap->getFromDB($_SESSION['ldap_server']);
 
+        $filter_name1 = "condition";
+        $filter_name2 = "condition";
```
Comment (trasher) [inline src/AuthLDAP.php:1462]: Not sure for this one. With your change, you always reach the `BOTH` case. I guess you should let the original code in the switch as is, and declare both filter variables to null per default.

Comment (cedric-anne) [reply src/AuthLDAP.php]: done

File: src/DBmysql.php [hunk 2]
```diff
@@ -363,10 +363,9 @@ public function query($query)
             $SQL_TOTAL_REQUEST++;
             $DEBUG_SQL["queries"][$SQL_TOTAL_REQUEST] = $query;
         }
-        if ($is_debug || $this->execution_time === true) {
-            $TIMER = new Timer();
-            $TIMER->start();
-        }
+
```
Comment (trasher) [inline src/DBmysql.php:366]: I do not really understand why this is no longer under a condition. Just for `$TIMER` to be defined?

File: src/DropdownTranslation.php [hunk 3]
```diff
@@ -481,11 +481,15 @@ public static function showTranslations(CommonDropdown $item)
      */
     public function showForm($ID = -1, array $options = [])
     {
+        if (!isset($options['parent']) || !($options['parent'] instanceof CommonDBTM)) {
```
Comment (trasher) [inline src/DropdownTranslation.php:484]: Sounds like a logic change here; and we'll never know why nothing is displayed. At least, an warning message should be emitted in logs

File: src/Plugin.php [hunk 4]
```diff
@@ -1071,7 +1071,7 @@ public function activate($ID)
                 $this->unload($this->fields['directory']);
 
                 Session::addMessageAfterRedirect(
-                    sprintf(__('Plugin configuration must be done, it cannot be activated.') . ' ' . $msg, $this->fields['name']),
+                    sprintf(__('Plugin configuration must be done, it cannot be activated.'), $this->fields['name']),
```
Comment (trasher) [inline src/Plugin.php:1074]: Either `sprintf()` is useless, either string miss a replacement character. Note this is the same for at least line 1010

File: src/DBmysql.php (hunk 2 above)
Comment (trasher) [reply src/DBmysql.php:366]: (Looking at `Timer` class, this should not really impact)

Comment (cedric-anne) [reply src/DBmysql.php:366]: > I do not really understand why this is no longer under a condition. Just for `$TIMER` to be defined?

PHPStan is not able to recognize that `$TIMER` is always defined where it is used.

File: src/DropdownTranslation.php (hunk 3 above)
Comment (cedric-anne) [reply src/DropdownTranslation.php:484]: `$item` is always used on lines 507/508, so method would have trigger a fatal error if call without a valid object in `$options['parent']`.

I added a `trigger_error` to emit something in logs.

File: src/Plugin.php (hunk 4 above)
Comment (cedric-anne) [reply src/Plugin.php]: Indeed, I fixed it.

**Review by trasher — APPROVED** [review]
LGTM

**Pull request merged**